eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelData.scala

```diff
@@ -563,10 +564,9 @@ case class LocalParams(nodeId: PublicKey,
                        toSelfDelay: CltvExpiryDelta,
                        maxAcceptedHtlcs: Int,
                        isInitiator: Boolean,
-                       defaultFinalScriptPubKey: ByteVector,
+                       defaultFinalScriptPubKey: Option[ByteVector],
```

Comment:
You are not taking advantage of the semantic change to actually remove the ambiguity. Also we should use our usual naming convention for options:
```suggestion
                       upfrontShutdownScript_opt: Option[ByteVector],
```

`RemoteParams.shutdownScript` should be renamed as well.

Reply:
done in https://github.com/ACINQ/eclair/pull/2565/commits/21a46ca164a11a5954b87d4569fc77b30954ef0a